auth/providers/azure/graph/obo_tokenprovider.go

```diff
@@ -74,6 +75,9 @@ func (u *oboTokenProvider) Acquire(ctx context.Context, token string) (AuthRespo
 		klog.V(10).Infoln(cmd)
 	}
 
+	if c, ok := ctx.Value(oauth2.HTTPClient).(*http.Client); ok {
+		u.client = c
```

Comment:
This shouldn't modify the token provider (`u.client`). Store a reference to the HTTP client in a variable and modify it there.

The issue is if this is called twice (first with a context that holds an HTTP client, then with a context that has no HTTP client), the second call will use the HTTP client of the first.

Reply:
removed this part to make the pr focus on authz.